src/module.c

```diff
@@ -3048,6 +3051,264 @@ int RM_HashGet(RedisModuleKey *key, int flags, ...) {
     return REDISMODULE_OK;
 }
 
+/* --------------------------------------------------------------------------
+ * Key API for the stream type.
+ * -------------------------------------------------------------------------- */
+
+/* Parses a stream ID. Returns REDISMODULE_OK and updates `id_parsed` on success
+ * and returns REDISMODULE_ERR on syntax error. The special IDs "+" and "-" are
+ * allowed.
+ *
+ * RedisModuleStreamID is a struct with two 64-bit fields, which is used in
+ * stream functions and defined as
+ *
+ *     typedef struct RedisModuleStreamID {
+ *         uint64_t ms;
+ *         uint64_t seq;
+ *     } RedisModuleStreamID;
+ */
+int RM_StreamParseID(RedisModuleString *id_str, RedisModuleStreamID *id_parsed) {
+    streamID id;
+    if (streamParseID(id_str, &id) == C_OK) {
+        id_parsed->ms = id.ms;
+        id_parsed->seq = id.seq;
+        return REDISMODULE_OK;
+    } else {
+        return REDISMODULE_ERR;
+    }
+}
+
+/* Formats a stream ID as a string. The returned string must be freed with
+ * RedisModule_FreeString(), unless automatic memory is enabled.
+ *
+ * The passed context `ctx` may be NULL if necessary. See the
+ * RedisModule_CreateString() documentation for more info. */
+RedisModuleString *RM_StreamFormatID(RedisModuleCtx *ctx, RedisModuleStreamID *id) {
```

Comment:
maybe RM_CreateStringFromStreamID?

Reply:
Good idea. I wanted to start them all with `RM_Stream`, but this one is actually just like `RedisModule_CreateStringFrom(LongLong|Double|...type...)` so I like it.

Should I put it together with the other `CreateStringFrom*` functions so they show up together in the ref docs?

Reply:
yes please

Reply:
Also renaming the parse function to `RM_StringToStreamID` (matching `StringToLongLong` and others), OK?

Reply:
yes please